tensorbay/cli/log.py

```diff
@@ -76,19 +102,14 @@ def _implement_log(  # pylint: disable=too-many-arguments
         revisions = (
             [tbrn_info.revision] if tbrn_info.revision else [dataset_client.status.branch_name]
         )
-
-    Printer: Union[Type[_GraphPrinter], Type[_CommitPrinter]] = (
-        _GraphPrinter if graph else _CommitPrinter
-    )
-    commit_generator = islice(
-        Printer(dataset_client, revisions, commit_id_to_branches, oneline).generate_commits(),
+    Printer: Union[Type[_GraphPrinter], Type[_Printer]] = _GraphPrinter if graph else _Printer
+    log_generator = islice(
+        Printer(
+            dataset_client, revisions, commit_id_to_branches, oneline, show_drafts
+        ).generate_commits_and_drafts(),
         max_count,
     )
-    if sys.platform.startswith("win"):
-        for item in commit_generator:
-            click.echo(item)
-    else:
-        click.echo_via_pager(commit_generator)
+    _echo_logs(log_generator)
```

Comment:
why we need to add function `_echo_log`?

Reply:
To avoid pylint `too-many-locals`  error